fuel-vm/src/interpreter/initialization.rs

```diff
@@ -38,8 +43,13 @@ where
 
         let tx_size = self.transaction().serialized_size() as Word;
 
-        self.registers[REG_GGAS] = self.transaction().limit();
-        self.registers[REG_CGAS] = self.transaction().limit();
+        let gas = self
+            .transaction()
+            .limit()
+            .checked_sub(gas_used_by_predicates)
+            .expect("Bug! Predicates have used more gas than allowed");
```

Comment:
Use bug error type

Reply:
Also since the VM doesn't check whether the predicates flag is checked, I don't think this is necessarily a bug/panic. It could just be a regular error.